src/Hedgehog/Plutus/TestData.hs

```diff
@@ -138,26 +140,63 @@ instance (Integral i) => TestData (ShouldBeNatural i) where
 
   generalise = MightBeNegative . fromIntegral
 
-newtype Good' a = Good {unGood' :: Good a}
+newtype Good' a = G {unGood' :: Good a}
 
 deriving stock instance (Eq (Good a)) => Eq (Good' a)
 deriving stock instance (Show (Good a)) => Show (Good' a)
 
+newtype HKD d = HKD (d I)
```

Comment:
Afaict there's no reason to have this newtype. I've removed it on `brian/ditch-newtype` and it seems to be fine.

Reply:
It's for deriving via.

Reply:
Oh that makes sense, sorry about that.

Reply:
It's not considered good practice to declare instances for top-level type variables.